scripts/manage-db.py

```diff
@@ -13,22 +14,74 @@
 
 from auslib.db import AUSDatabase
 
+def cleanup_releases(trans, nightly_age, dryrun=True):
+    # This and the subsequent queries use "%%%%%" because we end up going
+    # through two levels of Python string formatting. The first is here,
+    # and the second happens at a low level of SQLAlchemy when the transaction
+    # is being executed.
+    query = """
+LEFT OUTER JOIN releases_history USING (name)
+LEFT JOIN rules ON (name=mapping)
+WHERE name LIKE '%%%%nightly%%%%'
+AND (rules.whitelist <> releases.name OR rules.whitelist IS NULL)
```

Comment:
It looks like we're trying to leave alone any release which is referenced as an active whitelist, but the the table join only considers mappings. Maybe <tt>LEFT JOIN rules ON (name=mapping OR name=whitelist)</tt>. Fairly edgecase, would need a whitelist with nightly in the name.